uikit/views/src/main/java/com/sygic/maps/uikit/views/searchresultlist/SearchResultList.kt

```diff
@@ -0,0 +1,106 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.sygic.maps.uikit.views.R
+import com.sygic.maps.uikit.views.common.AdvanceInfoView
+import com.sygic.maps.uikit.views.common.EmptyRecyclerView
+import com.sygic.maps.uikit.views.databinding.LayoutSearchResultListInternalBinding
+import com.sygic.maps.uikit.views.searchresultlist.adapter.ResultListAdapter
+
+/**
+ * A [SearchResultList] can be used as an visual presentation component for the search result items. It contains
+ * [EmptyRecyclerView] and pre-customized [AdvanceInfoView] component.
+ *
+ * TODO: MS-5681
+ */
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class SearchResultList @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.searchResultListStyle,
+    defStyleRes: Int = R.style.SygicSearchResultListStyle // TODO: MS-5681
+) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
+
+    private val binding: LayoutSearchResultListInternalBinding =
+        LayoutSearchResultListInternalBinding.inflate(LayoutInflater.from(context), this, true)
+
+    init {
+        binding.searchResultListRecyclerView.setHasFixedSize(true)
+        binding.searchResultListRecyclerView.layoutManager = LinearLayoutManager(context)
+    }
+
+    /**
+     * Set a new adapter to provide [RecyclerView] child views on demand.
+     *
+     * When adapter is changed, all existing views are recycled back to the pool. If the pool has
+     * only one adapter, it will be cleared.
+     *
+     * @param adapter The new adapter to set, or null to set no adapter.
+     */
+    fun <T : Parcelable> setAdapter(adapter: ResultListAdapter<T, ResultListAdapter.ItemViewHolder<T>>) {
+        binding.searchResultListRecyclerView.adapter = adapter
+    }
+
+    /**
+     * Add a listener that will be notified of any changes in scroll state or position.
+     *
+     * Components that add a listener should take care to remove it when finished.
+     * Other components that take ownership of a view may call [clearOnScrollListeners]
+     * to remove all attached listeners.
+     *
+     * @param listener listener to set
+     */
+    fun addOnScrollListener(listener: RecyclerView.OnScrollListener) {
+        binding.searchResultListRecyclerView.addOnScrollListener(listener)
+    }
+
+    /**
+     * Remove a listener observing changes.
+     *
+     * Components that add a listener should take care to remove it when finished.
+     * Other components that take ownership of a view may call [clearOnScrollListeners]
+     * to remove all attached listeners.
+     *
+     * @param listener listener to remove
+     */
+    fun removeOnScrollListener(listener: RecyclerView.OnScrollListener) {
+        binding.searchResultListRecyclerView.removeOnScrollListener(listener)
+    }
+
+    /**
+     * Remove all secondary listener that were notified of any changes in scroll state or position.
```

Comment:
secondary?
```suggestion
     * Remove all listeners which are registered for scroll state or position changes.
```

Reply:
to je zkopírované z RecyclerView clearOnScrollListeners() metody.
![image](https://user-images.githubusercontent.com/4901251/59764682-90004100-929c-11e9-9b4c-d83599c3b6a8.png)

Reply:
kks to fakt? okrem toho ze nechapem co chcu povedat tym secondary, tak all `listener` bez `s` asi nebude gramaticky spravne

Reply:
jj, po pravdě jsem to jen zkopíroval, ani jsem to nečetl. ani ve snu by mě nenapadlo, že tam budou mít takovou blbost. příště budu už pozornější ;)